### PR TITLE
Enforce even indentation and automate some conventions

### PR DESCRIPTION
**Pull request by fredrik-bakke**
Wraps up the last PR on pre-commit stuff.

### Summary
- add detection for empty sections at the end of a file
- add autoremoval of empty code blocks
- add autoremoval of punctuation at the end of headers
- fix all uneven indentation in the library
- enable enforcing of even indentation
- pre-commit now properly fails when a misplaced top-level header is detected

### CODE REVIEW

File: src/univalent-combinatorics/orientations-complete-undirected-graph.lagda.md [hunk 1]
```diff
@@ -2439,136 +2439,137 @@ module _
...
+                    ( map-inv-equiv
+                      ( transposition
+                        ( standard-2-Element-Decidable-Subtype
+                          ( has-decidable-equality-count eX)
+                          ( np)))))
+                  ( x)))
+            ( P)
+            ( inl
               ( inv
-                ( right-computation-standard-transposition
```
Comment (fredrik-bakke) [inline src/univalent-combinatorics/orientations-complete-undirected-graph.lagda.md:2555]: Did I override a renaming here?

Comment (EgbertRijke) [reply src/univalent-combinatorics/orientations-complete-undirected-graph.lagda.md:2555]: I don't know, but it looks like `right-computation-standard-transposition` and `is-fixed-point-standard-transposition` do the same thing, which means that they should be merged.

**Comment by EgbertRijke** [comment]
How confident are you that you increased/decreased the indentation level of the oddly indented code correctly?:)

**Comment by fredrik-bakke** [comment]
I did go through all the changes a second time, but I won't guarantee that everything is correct now. The main goal wasn't to make everything correct anyways, but to avoid introducing more mistakes later.

**Pull request merged**